openvswitch: 2.5.4 -> 2.12.0

netixx opened the pull request:
Openvswitch was upgraded to the latest
stable version (currenty 2.9.2). This remove ovs-monitor-ipsec
commands.

LTS version is still available using
`config.virtualisation.vswitch.package = pkgs.openvswitch-lts`

This commit is a split from the original PR #35127.

###### Motivation for this change

Update openvswitch to a more recent version.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

netixx committed:
openvswitch: 2.5.4 -> 2.12.0

Openvswitch was upgraded to the latest
stable version (currenty 2.12.0). This remove ovs-monitor-ipsec
commands.

LTS version is still available using
`config.virtualisation.vswitch.package = pkgs.openvswitch-lts`
it has been upgraded to 2.5.6.

This commit is a split from the original PR #35127.

netixx commented:
Bumped version to 2.10.1 and 2.5.6. Rebased on current master

netixx commented:
@GrahamcOfBorg build openvswitch-lts
@GrahamcOfBorg build openvswitch

offlinehacker commented on pkgs/os-specific/linux/openvswitch/default.nix:
```diff
@@ -36,27 +37,13 @@ in stdenv.mkDerivation rec {
   '';
 
   enableParallelBuilding = true;
   doCheck = false; # bash-completion test fails with "compgen: command not found"
```
What if we disable that test?

netixx replied:
I just replaced python in the previous code. I guess there was work to try make the test run but in was aborted in the end... Should we comment out the tests if someone wants to pickup the work later or delete the whole thing ?

offlinehacker commented on pkgs/os-specific/linux/openvswitch/lts.nix:
```diff
@@ -0,0 +1,77 @@
+  '';
+
+  enableParallelBuilding = true;
+  doCheck = false; # bash-completion test fails with "compgen: command not found"
```
The same here, since above we are fixing tests and we are disabling them here?

offlinehacker commented on nixos/modules/virtualisation/openvswitch.nix:
```diff
@@ -42,6 +42,9 @@ in {
       default = false;
       description = ''
         Whether to start racoon service for openvswitch.
+        Supported only if openvswitch version is less than 2.6.0.
```
Did they drop support in later versions?

netixx replied:
Yes they did: https://github.com/openvswitch/ovs/commit/2b02d770c4cb381ec32cd4b7b1e991c42b448884

offlinehacker reviewed (commented):
Everything else looks good to me, thanks!

offlinehacker commented:
It's on my agenda for tomorrow to check this pull request and preferably merge it 🙂

netixx commented:
Rebased to master and resolved conflicts.

I can put myself as a maintainer if this helps!

netixx commented:
Updated to 2.12.0 (support for 4.19 and 4.20 kernels, which are in use in nixos).
Also added myself as maintainer.

kmcopper4 commented:
I would really appreciate this being merged!

kmcopper committed:
maintainers: add kmcopper

kmcopper committed:
openvswitch: many improvements

- tidy maintainer definition
- add kmcopper to maintainers
- use https for url fetch
- update lts to 2.5.9
- compile with openssl 1.1
- use https for homepage url
- use python3 in 2.12

netixx committed:
Merge pull request #1 from kmcopper/openvswitch-improvements

openvswitch improvements

netixx commented:
Updated with pull request https://github.com/netixx/nixpkgs/pull/1

kmcopper reviewed (approved):
Looks good to me. Feel free to rebase on master as needed. Appreciate the merge and hope to get this accepted. Should we plan to backport this to 19.09 and default to the LTS variant for 19.09 since that is the version there currently?

kmcopper commented:
@offlinehacker we would greatly appreciate a merge here. Thanks!

fpletz reviewed (approved):
Looks fine. Thanks!
